### DISCUSSION TITLE
Use a connected node(not lighthouse) as DNS for other clients?

### DISCUSSION THREAD

**Discussion by zabihexes** [DefinedNet/nebula-docs#94] [Q&A] [answered]
Is it possible to set nebula app to use a specific node as DNS server when connecting via nebula.

For example if I connect using nebula app on my phone then it would use a dns from another node connected to nebula.

My use case would be having adguard server return different IP depending on if I am connecting via lan or nebula or tailscale....
This way I could use the same domain no matter how I am connecting to the device.

**Comment by zabihexes** [comment]
Tailscale supports this kind of thing for example.

**Comment by johnmaguire** [comment, marked as answer]
@Yaikava Hey there - thanks for the ask.

Today, Nebula does not support configuring DNS resolvers for a host. The challenge on desktop devices is that there are many ways DNS resolvers are configured on systems and it's difficult to implement a solution which satisfies all of them. Nebula aims to keep a well-defined scope focusing on the tunnel, and so the suggestion here would be to write some sort of wrapper (e.g. using systemd dependencies or a bash script) which configures the resolvers for your needs. (See https://github.com/slackhq/nebula/issues/318#issuecomment-1355427377)

On the mobile side the situation is a bit different because there are only two main platforms to support (Android and iOS), and it's not as feasible to script a solution like you can on desktop. This does require development work however. If this would be useful to you, please consider adding your vote to https://github.com/DefinedNet/mobile_nebula/issues/9

Thanks for using Nebula!
